rules/rules.go

```diff
@@ -15,10 +24,84 @@ func (ctx *ExecContext) Clone() *ExecContext {
 		Args:         slices.Clone(ctx.Args),
 		Package:      ctx.Package,
 		PackageIndex: ctx.PackageIndex,
+		GoVersion:    ctx.GoVersion,
 	}
 }
 
 type Rule interface {
 	Apply(ctx *ExecContext) error
 	Name() string
 }
+
+// GoVersion represents a Go version with major, minor, and patch components
+type GoVersion struct {
+	Major int
+	Minor int
+	Patch int
+}
+
+// Compare returns -1 if v < other, 0 if v == other, 1 if v > other
+func (v GoVersion) Compare(other GoVersion) int {
+	if v.Major != other.Major {
+		if v.Major < other.Major {
+			return -1
+		}
+		return 1
+	}
+	if v.Minor != other.Minor {
+		if v.Minor < other.Minor {
+			return -1
+		}
+		return 1
+	}
+	if v.Patch != other.Patch {
+		if v.Patch < other.Patch {
+			return -1
+		}
+		return 1
+	}
+	return 0
+}
+
+// IsAtLeast returns true if this version is >= the specified version
+func (v GoVersion) IsAtLeast(major, minor, patch int) bool {
```

Comment:
Sorry for repeated comments.

All methods should have the common interface: GoVersion or tuple (major, minor, patch).
Compare has GoVersion, while IsAtLeast has the tuple.
I'd like to resolve the inconsistency by fixing IsAtLeast, adding methods for both ways, or removing IsAtLeast.

```suggestion
func (v GoVersion) IsAtLeast(other GoVersion) bool {
```

Reply:
fixed in bd3123f